dynparquet/schema.go

```diff
@@ -1350,47 +1351,67 @@ func mergeDynamicRowGroupDynamicColumns(rowGroups []DynamicRowGroup) map[string]
 	return MergeDynamicColumnSets(sets)
 }
 
-func MergeDynamicColumnSets(sets []map[string][]string) map[string][]string {
-	dynamicColumns := map[string][][]string{}
-	for _, set := range sets {
-		for k, v := range set {
-			_, seen := dynamicColumns[k]
-			if !seen {
-				dynamicColumns[k] = [][]string{}
-			}
-			dynamicColumns[k] = append(dynamicColumns[k], v)
-		}
-	}
+func MergeDynamicColumnSets(sets []map[string][]string) (o map[string][]string) {
+	m := newDynamicColumnSetMerger()
+	o = m.Merge(sets)
+	m.Release() // avoid defer penalty by releasing manually
```

Comment:
defer is pretty cheap in these new versions of go, have you seen it be a bottleneck? I would just `defer` this to also execute on e.g. panic (if that is a possiblity)

Reply:
> have you seen it be a bottleneck?

nope, the overhead is negligible.  There is no panic ,so we don't need to recover.

done f05d317